src/AppBundle/Entity/Report/Report.php

```diff
@@ -712,21 +721,41 @@ public function getStatus()
     }
 
     /**
-     * @return ArrayCollection
+     * @return ArrayCollection|Document[]
      */
     public function getDocuments()
     {
         return $this->documents;
     }
 
     /**
-     * @param ArrayCollection $documents
-     *
-     * @return $this
+     * @param Document $document
      */
-    public function setDocuments($documents)
+    public function addDocument(Document $document)
     {
-        $this->documents = $documents;
-        return $this;
+        if (!$this->documents->contains($document)) {
+            $this->documents->add($document);
+        }
     }
```

Comment:
I left it on purpose to keep it simple, as not needed. Doctrine already creates the setters in the proxies when missing. JMS doesn't need them as the API as itonly reads those documents, and never sets them in block (unless when created manually). JMS for setter in general is not required in the API as we set them manually in the endpoint.
I can still add the setters, or maybe - if we want it for all the entities - we create a ticket to go through all the entities ?